src/DatePicker/index.jsx

```diff
@@ -7,70 +7,379 @@
+import moment                       from 'moment';
+import _                            from 'lodash';
 
+import copy                         from './copy.json';
 import DatePickerItem               from './DatePickerItem';
 import DatePickerHeader             from './DatePickerHeader';
 import { attachEvents, useTheme }   from '../utils';
 
 
+const DAY_LABELS = _.range( 0, 7 ).map( day => ( {
+    label : copy.dayHeaders[ day ],
+} ) );
+
+/**
+ * returns the timestamp of the current moment
+ *
+ * @return {Number} timestamp
+ */
+function now()
+{
+    return new Date().getTime();
+}
+
+/**
+ * returns utc of the timestamp passed
+ *
+ * @param {Number} timestamp passed
+ *
+ * @return {Number} UTC timestamp
+ */
+function $m( timestamp )
+{
+    return moment( timestamp ).utc();
+}
+
+/**
+ * checks if 2 timestamp are equal
+ *
+ * @param {Number}  ts1 timestamp
+ * @param {Number}  ts2 timestamp
+ * @param {String}  precision precision to compare
+ *
+ * @return {Boolean}
+ */
+function isTimestampEqual( ts1, ts2, precision )
+{
+    return $m( ts1 ).isSame( $m( ts2 ), precision );
+}
+
+/**
+ * Timestamp conversion to Human time ( hours )
+ *
+ * @param {Number}  timestamp timestamp
+ *
+ * @return {String} human readable time ( hours )
+ */
+function formatHours( timestamp )
+{
+    if ( !_.isNumber( timestamp ) ) return '';
+    return $m( timestamp ).format( 'HH' );
+}
+
+/**
+ * Timestamp conversion to Human time ( minutes )
+ *
+ * @param {Number}  timestamp timestamp
+ *
+ * @return {String} human readable time ( minutes )
+ */
+function formatMinutes( timestamp )
+{
+    if ( !_.isNumber( timestamp ) ) return '';
+    return $m( timestamp ).format( 'mm' );
+}
+
+
+const useTimestamp = ( defaultValue = now(), value ) =>
+{
+    const [ timestamp, setTimestamp ] = useState( defaultValue );
+
+    const setter = ( newValue ) =>
+    {
+        if ( value === undefined )
+        {
+            setTimestamp( newValue );
+        }
+    };
+
+    return [ value || timestamp, setter ];
+};
+
+
 const componentName = 'DatePicker';
 
 const DatePicker = props =>
 {
+    const [ timestamp, setTimestamp ] = useTimestamp(
+        props.defaultValue,
+        props.value,
+    );
+    const [ gridStartState, setGridStartState ] = useState( null );
+    const [ hourValue, setHourValue ] = useState( undefined );
+    const [ minuteValue, setMinuteValue ] = useState( undefined );
+
     const cssMap = useTheme( componentName, props );
 
+
+    const gridStartTimestamp = gridStartState || $m( timestamp )
+        .startOf( props.type === 'month' ? 'year' : 'month' )
+        .valueOf();
+
+    const isUnitSelectable = (
+        itemTimestamp,
+        unit,
+    ) =>
+    {
+        const { max } = props;
+        const min = props.min || now();
+
+        if ( itemTimestamp > max ) return false;
+
+        return $m( itemTimestamp ).add( 1, unit ) > min;
+    };
+
+
+    const dayMatrix = () =>
```

Comment:
I don’t think this should be a function. I think we can just do:
```
const dayMatrix = _.chunk( days, 7 );
```
at the bottom